app/views/assets/_tags_box.html.erb

```diff
@@ -1,5 +1,8 @@
 <%
-   entity=controller_name.singularize
+   # Make this partial work when calling from views of different controllers by passing 'ctrl_name' param
+   ctrl_name||=nil
+   resource_id||=nil
+   entity=ctrl_name&.singularize||controller_name.singularize
```

Comment:
I think all these things are a consequence of the decision to render all types of views from the same controller.

I think you would probably find things easier if you mixed in a single action into each asset type's controller.

We do this for previews:
https://github.com/seek4science/seek/blob/09f6b68b8a33bae4bacf733f94af94dc8ad728c3/lib/seek/preview_handling.rb